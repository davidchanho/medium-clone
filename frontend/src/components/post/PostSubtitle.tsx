import React from "react";
import Card from "react-bootstrap/esm/Card";
import { clampText } from "../../helpers/clampText";
import { IPost } from "../../types";

function PostSubtitle({ ...post }: IPost) {
  const renderSubtitle = clampText(post.body, 50);

  return (
    <Card.Subtitle className="text-capitalize text-secondary">
      {renderSubtitle}
    </Card.Subtitle>
  );
}

export default PostSubtitle;
