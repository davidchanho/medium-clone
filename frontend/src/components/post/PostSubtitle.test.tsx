import React from "react";
import renderer from "react-test-renderer";
import { testPost } from "../../tests";
import PostSubtitle from "./PostSubtitle";

describe("PostSubtitle", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PostSubtitle {...testPost} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
