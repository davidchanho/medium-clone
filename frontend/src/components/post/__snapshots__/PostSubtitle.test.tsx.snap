// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSubtitle renders correctly 1`] = `
<div
  className="text-capitalize text-secondary card-subtitle h6"
>
  test...
</div>
`;
